pandas/tests/io/test_common.py

```diff
@@ -411,3 +414,298 @@ def test_is_fsspec_url():
+        zipper = gzip.GzipFile(fileobj=bio, mode="w")
+        zipper.write(response_bytes)
+        zipper.close()
+        response_bytes = bio.getvalue()
+        return response_bytes
+
+    def write_back_bytes(self, response_bytes):
+        """
+        shared logic at the end of a GET request
+        """
+        self.wfile.write(response_bytes)
+
+
+class CSVUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+
+        self.send_header("Content-Type", "text/csv")
+        self.end_headers()
+
+        response_bytes = response_df.to_csv(index=False).encode("utf-8")
+        self.write_back_bytes(response_bytes)
+
+
+class GzippedCSVUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "text/csv")
+        self.send_header("Content-Encoding", "gzip")
+        self.end_headers()
+
+        response_bytes = response_df.to_csv(index=False).encode("utf-8")
+        response_bytes = self.gzip_bytes(response_bytes)
+
+        self.write_back_bytes(response_bytes)
+
+
+class JSONUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/json")
+        self.end_headers()
+
+        response_bytes = response_df.to_json().encode("utf-8")
+
+        self.write_back_bytes(response_bytes)
+
+
+class GzippedJSONUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/json")
+        self.send_header("Content-Encoding", "gzip")
+        self.end_headers()
+
+        response_bytes = response_df.to_json().encode("utf-8")
+        response_bytes = self.gzip_bytes(response_bytes)
+
+        self.write_back_bytes(response_bytes)
+
+
+class ParquetPyArrowUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/octet-stream")
+        self.end_headers()
+
+        response_bytes = response_df.to_parquet(index=False, engine="pyarrow")
+
+        self.write_back_bytes(response_bytes)
+
+
+class ParquetFastParquetUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/octet-stream")
+        self.end_headers()
+
+        # the fastparquet engine doesn't like to write to a buffer
+        # it can do it via the open_with function being set appropriately
+        # however it automatically calls the close method and wipes the buffer
+        # so just overwrite that attribute on this instance to not do that
+        def dummy_close():
+            pass
+
+        bio = BytesIO()
+        bio.close = dummy_close
```

Comment:
do you need to do this in a mock? / monkeypatch?

Reply:
I found that fsspec allowed for "writing to a file" in memory.  I changed to using that for creating the fast parquet bytes reponse which avoids any patching/mocking here.